### ISSUE TITLE
Building CKEditor fails due to a webpack-source issue

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#658]
## 🐞 Is this a bug report or feature request? (choose one)

- Bug report

## 📋 Steps to reproduce

1. `git clone <one-of-the-builds>`
2. `cd <one-of-the-builds> && npm i`
3. `npm run build`

## ✅ Expected result

All works fine

## ❎ Actual result

```
/workspace/ckeditor5/node_modules/webpack-sources/node_modules/source-map/lib/source-map-generator.js:276
        throw new Error(
        ^

Error: original.line and original.column are not numbers -- you probably meant to omit the original mapping entirely and only map the generated position. If so, pass null for the original mapping instead of an object with empty or null values.
    at SourceMapGenerator_validateMapping [as _validateMapping] (/workspace/ckeditor5/node_modules/webpack-sources/node_modules/source-map/lib/source-map-generator.js:276:15)
```

Related webpack's issues:

* https://github.com/webpack/webpack/issues/5931
* https://github.com/webpack/webpack-sources/pull/27#issuecomment-341891811
* https://github.com/mozilla/source-map/issues/304

tl;dr is that `webpack-sources@1.0.2` has a regression and every `webpack@3.*.*` version will unfortunately install it.

## 📃 Other details that might be useful

There's a workaround – you need to force npm to install `webpack-sources@1.0.1`. So it's enough to add this line to CKE build's `package.json`:

```
diff --git a/package.json b/package.json
index 1dd7a09..a00b739 100644
--- a/package.json
+++ b/package.json
@@ -98,7 +98,8 @@
     "style-loader": "^0.18.2",
-    "webpack": "^3.5.5"
+    "webpack": "^3.5.5",
+    "webpack-sources": "1.0.1"
```

**Comment by Reinmar** [comment]
This is upstream's issue but let's monitor it. Also, the workaround may be helpful for someone.

**Comment by Reinmar** [comment]
OK, I decided to hardcode webpack-sources to 1.0.1 in all builds because we'll be releasing them today and it's still broken.

Let's revert the change when everything will be fine.

**Comment by oleq** [comment]
> OK, I decided to hardcode webpack-sources to 1.0.1 in all builds because we'll be releasing them today and it's still broken.
> 
> Let's revert the change when everything will be fine.

This is what I suggested. Because of this webpack issue, our builds are broken for early adopters and this is not a good thing for us when our software fails to build. Not a good impression at all.

**Comment by phoenixwong** [comment]
Hi there,

Sadly, the `" webpack-sources": "1.0.1"` doesn't work for me.

I tried **classic** build and **inline** build (stable) today, with`"webpack-sources": "1.0.1"`, but still seeing the same `original.line and original.column are not numbers` error.

1. `git clone -b stable <one-of-the-builds>`
2. `cd <one-of-the-builds>`
3. `npm install`
4. `npm run build`
5. Error pops out.

FYI, I doubled checked the node_modules, it's using **source-map 0.5.7** and **webpack-sources 1.0.1**, under **webpack 3.10.0**

**Comment by ma2ciek** [comment]
Hi @phoenixwong,

That's sad, thanks for reporting the issue.

What version of the NPM do you use? I'll try to reproduce it.

**Comment by ma2ciek** [comment]
I followed the installation steps using `NPM@4.6.1` and everything worked fine.

That's the important part of the `npm list --depth 1`.
<img width="277" alt="screen shot 2018-01-05 at 12 01 32" src="https://user-images.githubusercontent.com/2768047/34606733-d2f2e770-f210-11e7-997d-089ab43d0c3a.png">

**Comment by Reinmar** [comment]
> Error pops out.

What kind of error, @phoenixwong?

**Comment by phoenixwong** [comment]
Hi guys,

Thanks for the prompt reply! Here's the error message when running classic build

```
npm run create-entry-file && npm run build-ckeditor

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.2 create-entry-file /workplace/ckeditor/ckeditor5-build-classic
> node bin/create-entry-file.js

Creating the entry file...
Done.

> @ckeditor/ckeditor5-build-classic@1.0.0-alpha.2 build-ckeditor /workplace/ckeditor/ckeditor5-build-classic
> sh bin/build-ckeditor.sh

Building 'build/ckeditor.js'...

/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:276
        throw new Error(
        ^

Error: original.line and original.column are not numbers -- you probably meant to omit the original mapping entirely and only map the generated position. If so, pass null for the original mapping instead of an object with empty or null values.
    at SourceMapGenerator_validateMapping [as _validateMapping] (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:276:15)
    at SourceMapGenerator_addMapping [as addMapping] (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:110:12)
    at /workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:72:17
    at Array.forEach (native)
    at SourceMapConsumer_eachMapping [as eachMapping] (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-consumer.js:157:14)
    at Function.SourceMapGenerator_fromSourceMap [as fromSourceMap] (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:48:24)
    at SourceMapSource.node (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/webpack-sources/lib/SourceMapSource.js:32:35)
    at /workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:59:50
    at Array.map (native)
    at ConcatSource.node (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/node_modules/webpack-sources/lib/ConcatSource.js:58:63)
    at ConcatSource.proto.sourceAndMap (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/node_modules/webpack-sources/lib/SourceAndMapMixin.js:30:18)
    at getTaskForFile (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/SourceMapDevToolPlugin.js:33:30)
    at chunk.files.forEach.file (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/SourceMapDevToolPlugin.js:91:21)
    at Array.forEach (native)
    at /workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/SourceMapDevToolPlugin.js:89:18
    at Array.forEach (native)
    at Compilation.<anonymous> (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/SourceMapDevToolPlugin.js:88:12)
    at Compilation.applyPlugins1 (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:75:14)
    at self.applyPluginsAsync.err (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/Compilation.js:670:11)
    at next (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:202:11)
    at Compilation.compilation.plugin (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/BannerPlugin.js:67:5)
    at next (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:204:14)
    at Compilation.<anonymous> (/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/dist/index.js:119:11)
    at Compilation.applyPluginsAsyncSeries (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:206:13)
    at self.applyPluginsAsync.err (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/Compilation.js:666:10)
    at Compilation.applyPluginsAsyncSeries (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:195:46)
    at sealPart2 (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/Compilation.js:662:9)
    at Compilation.applyPluginsAsyncSeries (/workplace/ckeditor/ckeditor5-build-classic/node_modules/tapable/lib/Tapable.js:195:46)
    at Compilation.seal (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/Compilation.js:605:8)
    at applyPluginsParallel.err (/workplace/ckeditor/ckeditor5-build-classic/node_modules/webpack/lib/Compiler.js:504:17)
```

I ran `npm list --depth 1` and it looks slightly different from @ma2ciek 's

![screen shot 2018-01-05 at 22 29 15](https://user-images.githubusercontent.com/6954306/34613234-137fb85c-f268-11e7-9b6d-deeb0efddae1.png)

I'm going to update to the latest `npm` and `node` and tried again. Let you know ASAP.

**Comment by phoenixwong** [comment]
I start thinking my issue may be more relevant to **yarn** instead of **webpack-sources** or npm version.

1. Firstly, I've updated to *node v8.9.4* and *npm 5.6.0*.

2. `git clone -b stable <one-of-the-builds>` to two different folders. (Let's call them **folder_a** and **folder_b**)

3. In folder_a, try running with pure **npm**

    - `cd folder_a`
    - `npm install`
    - `npm run build`

    It can successfully build without errors! Problem solved!

4. In folder_b, use **yarn**

    - `cd folder_b`
    - `yarn` or `yarn install`
    - `yarn run build` or `npm run build`

     It will still trigger the following error --

```
Building 'build/ckeditor.js'...

/workplace/ckeditor/ckeditor5-build-classic/node_modules/babel-minify-webpack-plugin/node_modules/source-map/lib/source-map-generator.js:276
        throw new Error(
        ^

Error: original.line and original.column are not numbers -- you probably meant to omit the original mapping entirely and only map the generated position. If so, pass null for the original mapping instead of an object with empty or null values.

...
```

I've been switched to yarn for a while. This is the first time I met building issue with yarn. I'll take pure npm way as the solution. Thank you for all your prompt response : )

**Comment by Reinmar** [comment]
I've just realised that something's seriously wrong if this is still failing with latest webpack-sources. I've mistakenly updated this dep to 1.1.0 and got this error. It's 2 months and, apparently, it wasn't fixed on upstream as we thought.

I'm going to investigate this a bit further tomorrow.

**Comment by Reinmar** [comment]
OK, it's now clear that this issue wasn't fixed yet.

The initial ticket (https://github.com/webpack/webpack/issues/5931) was about DX and is still open. 

There are 3 more tickets regarding this error itself:

* https://github.com/webpack/webpack/issues/6131
* https://github.com/mozilla/source-map/issues/304
* https://github.com/webpack/webpack-sources/issues/28

All 3 are still open. Apparently, we need to wait a bit longer.

**Comment by dPhantum** [comment]
This is still an open issue, just today installed new, and same results, is their no work-around after 2 months?

**Comment by Reinmar** [comment]
The workaround is still to have `"webpack-sources": "1.0.1"` in your `package.json` :(

**Comment by ma2ciek** [comment]
We will be able to solve it when we go with webpack@4 and UglifyJs plugin, but there're some [issues](https://github.com/ckeditor/ckeditor5-dev/issues/371) with that too.

**Comment by kelvinlouis** [comment]
I also experienced this issue. Do not use yarn as a package manager. After deleting the node_modules and installing the packages using npm the build finished successfully.
